src/strands_tools/mcp_client.py

```diff
@@ -0,0 +1,899 @@
+        """Get the wrapped tool name."""
+        return self._name
+
+    @property
+    def description(self) -> str:
+        """Get the tool description."""
+        return self._tool_spec.get("description", "")
+
+    @property
+    def tool_spec(self) -> ToolSpec:
+        """Get the tool specification."""
+        return self._tool_spec
+
+    @property
+    def tool_type(self) -> str:
+        """Get the tool type."""
+        return "mcp"
+
+    @property
+    def supports_hot_reload(self) -> bool:
+        """MCP tools don't support hot reload."""
+        return False
+
+    def invoke(self, tool: ToolUse, *args: Any, **kwargs: Any) -> ToolResult:
+        """Execute the MCP tool through the connection."""
+        tool_use_id = tool.get("toolUseId", "unknown")
+
+        try:
+            # Extract the input parameters
+            tool_input = tool.get("input", {})
+
+            logger.debug(
+                "Invoking MCP tool: connection_id=%s, tool_name=%s, input=%s",
+                self.connection_id,
+                self.original_tool_name,
+                tool_input,
+            )
+
+            # Call the tool using direct client
+            result = self.mcp_client.call_tool_sync(
+                tool_use_id=f"wrapper_{self.connection_id}_{self.original_tool_name}_{uuid.uuid4().hex[:8]}",
+                name=self.original_tool_name,
+                arguments=tool_input,
+            )
+
+            # Extract content from SDK's ToolResult format
+            if result.get("status") == "error":
+                error_msg = "Unknown error"
+                if result.get("content"):
+                    error_msg = result["content"][0].get("text", error_msg)
+                raise Exception(error_msg)
+
+            # Extract content
+            content = result.get("content", [])
+            if len(content) == 1 and "text" in content[0]:
+                result = content[0]["text"]
+            else:
+                result = content
+
+            # Format the result according to Strands expectations
+            if isinstance(result, dict) and "error" in result:
+                return {"toolUseId": tool_use_id, "status": "error", "content": [{"text": result["error"]}]}
+
+            # Convert MCP result to Strands format
+            content = []
+            if isinstance(result, str):
+                content.append({"text": result})
+            elif isinstance(result, list):
+                # If it's already a list of content items, use it directly
+                content = result
+            elif isinstance(result, dict):
+                # Handle structured results
+                if "content" in result:
+                    # Already in expected format
+                    content = result["content"]
+                else:
+                    # Convert dict to text representation
+                    import json
+
+                    content.append({"text": json.dumps(result, indent=2)})
```

Comment:
Images can be bytes which can not be json serialized, assuming will throw error here ^^